README.md

```diff
@@ -1,178 +1,251 @@
 
-In standalone mode`rouge` accepts the same parameters like
-`--MACHINE`, `--ENABLE_ANDROID` as `moulin` do.
+## Flash the board
 
-This XT product provides two images: `full` and `android_only`. Latter
-is available only when `--ENABLE_ANDROID=yes`.
+### loaders
+We use [rcar_flash](https://github.com/xen-troops/rcar_flash) for flashing loaders.
+Get the tool into some folder.
+```
+git clone https://github.com/xen-troops/rcar_flash.git
+```
+Turn on your board and run the following command
+```
+sudo ./rcar_flash.py flash -c -f -b h3ulcb_4x2 -s /dev/ttyUSB0 -p ${WORK_DIR}/yocto/build-domd/tmp/deploy/images/h3ulcb/firmware all
+```
 
-You can prepare image by running
+### u-boot env
+Start `minicom` on the host.
+Reboot the board and press any key to stop u-boot.
 
+Set MAC address according to the sticker on the Ethernet connector
 ```
-# rouge prod-cockpit-rcar.yaml --ENABLE_ANDROID=yes -i full
+setenv ethaddr __:__:__:__:__:__
 ```
 
-This will create file `full.img` in your current directory.
-
-Also you can write image directly to a SD card by running
+Set commands required for the proper work of the u-boot
+```
+setenv bootargs
+setenv ipaddr 192.168.5.22
+setenv serverip 192.168.5.75
+setenv initrd_high 0xffffffffffffffff
+setenv bootcmd run bootcmd_emmc
+setenv bootcmd_emmc 'run emmc_xen_load; run emmc_dtb_load; run emmc_kernel_load; run emmc_xenpolicy_load; run emmc_initramfs_load; bootm 0x48080000 0x84000000 0x48000000'
+setenv emmc_dtb_load 'ext2load mmc 1:1 0x48000000 xen.dtb; fdt addr 0x48000000; fdt resize; fdt mknode / boot_dev; fdt set /boot_dev device mmcblk0'
+setenv emmc_initramfs_load ext2load mmc 1:1 0x84000000 uInitramfs
+setenv emmc_kernel_load ext2load mmc 1:1 0x8a000000 Image
+setenv emmc_xen_load ext2load mmc 1:1 0x48080000 xen
+setenv emmc_xenpolicy_load ext2load mmc 1:1 0x8c000000 xenpolicy
+
+setenv bootcmd_bsp 'run bsp_set_args; run bsp_cmd'
+setenv bsp_cmd 'tftp 0x48080000 Image; tftp 0x48000000 r8a77951-ulcb-kf.dtb; booti 0x48080000 - 0x48000000'
+setenv bsp_set_args 'setenv bootargs ignore_loglevel clk_ignore_unused=1 rw root=/dev/nfs nfsroot=/srv/nfs,nfsvers=3 ip=${ipaddr}:${serverip}::255.255.255.0'
+
+saveenv flash_cluster 'tftp 0x48000000 cluster_wrapper.bin; erase 0x0A000000 +0x${filesize}; cp.b 0x48000000 0x0A000000 0x${filesize}'
+
+saveenv
+```
 
+### cluster firmware
+Copy the cluster binary to TFTP folder
+```
+sudo cp ${WORK_DIR}/yocto/build-cluster-wrapper/tmp/deploy/cluster_wrapper.bin /srv/tftp
+```
+On the board, in the u-boot, run the command
 ```
-# sudo rouge prod-devel-rcar.yaml --ENABLE_ANDROID=yes -i full -so /dev/sdX
+==> run flash_cluster
 ```
 
-**BE SURE TO PROVIDE CORRECT DEVICE NAME**. `rouge` have no
-interactive prompts and will overwrite your device right away. **ALL
-DATA WILL BE LOST**.
+### image
+Put `full.img` to the NFS root
+```
+cp ${WORK_DIR}/full.img /srv/nfs
+```
+In the u-boot, start BSP
+```
+==> run bootcmd_bootcmd_bsp
```

Comment:
Hi @rshym, You only have the 'bootcmd_bsp' command. I guess there is a typo here. Please fix it.

Reply:
Thanks for the rigorous reading! Fixed.